internal/services/containers/kubernetes_addons.go

```diff
@@ -235,6 +239,28 @@ func schemaKubernetesAddOnProfiles() *pluginsdk.Schema {
 						},
 					},
 				},
+				"azure_keyvault_secrets_provider": {
+					Type:     pluginsdk.TypeList,
+					MaxItems: 1,
+					Optional: true,
+					Elem: &pluginsdk.Resource{
+						Schema: map[string]*pluginsdk.Schema{
+							"enabled": {
```

Comment:
Are we sure that's a good idea? It would break convention with the other addon blocks, since they all have the "enabled" flag.

Additionally, for the sake of configuration, would be nice to be able to have the block in your manifests but keep it disabled, so enabling in the future just requires changing a variable.

Reply:
We would like to move away from having the enabled field for all AKS addons, but perhaps best to address this separately.